lib/tasks/bower-install.js

```diff
@@ -1,32 +1,48 @@
 'use strict';
 
 // Runs `bower install` in cwd
+const fs = require('fs-extra');
 const path = require('path');
 const existsSync = require('exists-sync');
 const execa = require('execa');
 const RSVP = require('rsvp');
 const Task = require('../models/task');
 const formatPackageList = require('../utilities/format-package-list');
 
+const logger = require('heimdalljs-logger')('ember-cli:tasks:bower-install');
+
 const Promise = RSVP.Promise;
 const resolve = RSVP.denodeify(require('resolve'));
+const writeJson = RSVP.denodeify(fs.writeJson);
 
 const cliPath = path.resolve(`${__dirname}/../..`);
 
 class BowerInstallTask extends Task {
 
+  resolveBower() {
+    logger.info('Resolving "bower" from %s ...', cliPath);
+    return resolve('bower', { basedir: cliPath });
+  }
+
+  importBower(path) {
+    logger.info('Importing "bower" from: %s', path);
+    this.bower = require(path);
+  }
+
   ensureBower() {
     if (this.bower) {
-      return Promise.resolve(this.bower);
+      return Promise.resolve();
     }
 
-    return resolve('bower', { basedir: cliPath })
+    return this.resolveBower()
       .catch(() => this.installBower()
-        .then(() => resolve('bower', { basedir: cliPath })))
-      .then(bowerPath => require(bowerPath));
+        .then(() => this.resolveBower()))
+      .then(bowerPath => this.importBower(bowerPath));
   }
 
   installBower() {
+    logger.info('Installing "bower" via NPM into: %s', cliPath);
```

Comment:
@Turbo87 We can

1. Check to see if there is a global, and if it is above our version. If so, we use it.
2. If no global, install one, return.
3. If global and under our version, install into the current `cliPath`.

Note: I'd prefer to always dump this into our top-level `ember-cli` instead of into the project's copy. You'll counter that version matters local to the version of CLI and I'd agree, especially for `bower` since we interact with it programmatically instead of shelling out, and we'll make no changes.